### PR TITLE
Don't try to use persistent compilation cache when running CPU computations.

### PR DESCRIPTION
**Pull request by skye**

### CODE REVIEW

File: jax/interpreters/xla.py [hunk 1]
```diff
@@ -67,14 +67,15 @@
 def compile_or_get_cached(backend, computation, compile_options):
     # Avoid import cycle between jax and jax.experimental
     from jax.experimental.compilation_cache import compilation_cache as cc
-    if cc.is_initialized():
-        cached_executable = cc.get_executable(computation, compile_options)
+    # Persistent compilation cache only implemented on TPU. Avoid errors on CPU.
+    if cc.is_initialized() and backend.platform != 'cpu':
```
Comment (zhangqiaorjc) [inline jax/interpreters/xla.py:71]: what about gpu?

can we narrow this to only supported platform ?

Comment (trevorcai) [reply jax/interpreters/xla.py:71]: GPU definitely fails to serialize the executable.

Comment (skye) [reply jax/interpreters/xla.py:71]: I intentionally did it this way so it does raise an error when using GPU, so people know it doesn't work. I think we may eventually wanna intercept all cache exceptions and print a warning but continue running, but for now I think it's better to fail. I could be convinced to just check for TPU though.

Comment (colemanliyah) [reply jax/interpreters/xla.py:71]: + 1 could we do backend.platform == 'tpu' ?

**Review by colemanliyah — APPROVED** [review]

File: jax/interpreters/xla.py (hunk 1 above)
Comment (skye) [reply jax/interpreters/xla.py]: Changed to platform == tpu.

**Review by zhangqiaorjc — APPROVED** [review]

**Pull request merged**